Documentation/flex-to-csi-migration.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Migration
+weight: 11900
+indent: true
+---
+
+# Flex to CSI Migration
+
+In Rook v1.8, the Flex driver has been deprecated. Before updating to v1.8, any Flex volumes created in previous versions of Rook will need to be converted to Ceph-CSI volumes.
+
+The tool [persistent-volume-migrator](https://github.com/ceph/persistent-volume-migrator) will help automate migration of Flex rbd volumes to Ceph-CSI volumes.
+
+## Migration Preparation
+
+1. Rook v1.7.x is required. If you have a previous version of Rook running, follow the [upgrade guide](https://rook.io/docs/rook/v1.7/ceph-upgrade.html) to upgrade from previous releases until on v1.7.x.
+2. Enable the CSI driver if not already enabled. See the [operator settings](https://github.com/rook/rook/blob/release-1.7/cluster/examples/kubernetes/ceph/operator.yaml#L29-L32) such as `ROOK_CSI_ENABLE_RBD`.
+3. Create the CSI storage class to which you want to migrate
+4. Existing volumes should be dynamically provisioned.
+ Show conversation
+5. Ceph-CSI storageClass should be created in which you want to migrate
+6. Create rbac required to run migrator pod
+   1) `kubectl create -f cluster/examples/kubernetes/ceph/flex-migration/rbac.yaml`
+7. Create migrator pod
+   1) `kubectl create -f cluster/examples/kubernetes/ceph/flex-migration/migrator.yaml`
+
+**NOTE**: The Migration procedure will come with a downtime; As we need to scale down the applications using the Volumes before migration.
+
+## Migrate a PVC
+   1) Stop the application pods that are consuming the flex volume(s) that need to be converted.
+   2) Connect to migration pod
+      1) `Migration_POD=$(kubectl -n rook-ceph get pod -l app=rook-ceph-migrator -o jsonpath='{.items[*].metadata.name}')`
+      2) `kubectl -n rook-ceph exec -it "$Migration_POD" -- sh`
+
+   3) Run command below to migrate single PVC
+      1) `<binary_name> --pvc=<pvc-name> --pvc-namespace=<pvc-namespace> --destinationstorageclass=<storageclass-name-to-migrate-in> --rook-namespace=<rook-ceph> --ceph-cluster-namespace=<ceph-cluster-namespace>`
```

Comment:
Let's move the params and sample output to a new section with its own title below. Then they can see all the steps before they read the detailed section.